### PR TITLE
Variabilize the /tmp path for uploaded files

### LINKED ISSUE #412: In several roles we use /tmp
Hi,
In several roles we use /tmp, it is preferable to use ansible tempfile. Sometimes  /tmp could not exists or badly reseized

Best regards

### PR DESCRIPTION
**Pull request by kpgtek**
#### Which issue(s) this PR fixes

Fixes #412

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp_vars_defaults/tdp-cluster/tdp-cluster.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@
 # Cluster-wide config #
 #######################
 
+# Variabilize the upload destintion directory (issue #412)
```
Comment (rpignolet) [inline tdp_vars_defaults/tdp-cluster/tdp-cluster.yml:9]: I propose this comment:
```suggestion
# Directory where TDP component releases are uploaded for each host
```
I think the issue ID is not needed.

**Comment by rpignolet** [comment]
Can you squash your commits and change `fix` with `feat` in commit message ? You add a variable, this is a new feature not a fix.

**Review by rpignolet — DISMISSED** [review]

**Review by mehdibn — APPROVED** [review]

File: tdp_vars_defaults/tdp-cluster/tdp-cluster.yml [hunk 2]
```diff
@@ -6,6 +6,9 @@
 # Cluster-wide config #
 #######################
 
+# Directory where TDP component releases are uploaded for each host 
+upload_tmp_dir: /tmp
```
Comment (rpignolet) [inline tdp_vars_defaults/tdp-cluster/tdp-cluster.yml:10]: Why using `tmp` in the variable name ? The directory where the releases will be uploaded does not need to be temporary. With this variable name it looks like the directory should be temporary.

Comment (kpgtek) [reply tdp_vars_defaults/tdp-cluster/tdp-cluster.yml:10]: I agree with your point of view, we need to find a relevant variable name.
The target directory could be a free choice for the user: temporary, transitory or archives.
Below, there are few suggestions:
- ```binaries_uploaded_dir```
- ```tdp_binaries_uploaded_dir```
- ```binaries_uploaded_target_dir```
- ```tdp_binaries_uploaded_target_dir```

Comment (PACordonnier) [reply tdp_vars_defaults/tdp-cluster/tdp-cluster.yml:10]: I'm fine `binaries_upload_dir`. The past tense "uploaded" does not feel right to me.

I think the question of prefixing the name with TDP will be answered in a future discussion where we will decide how to manage multiple TDP version ?

Comment (kpgtek) [reply tdp_vars_defaults/tdp-cluster/tdp-cluster.yml:10]: I think that "upload" is more suitable. So it remains 2 suggestions for the next step:
- ```binaries_upload_dir```
or
- ```tdp_binaries_upload_dir```

**Review by rpignolet — APPROVED** [review]

**Pull request merged**